src/Language/PureScript/Pretty/JS.hs

```diff
@@ -162,8 +162,10 @@ conditional = mkPattern match
 accessor :: (Emit gen) => Pattern PrinterState JS (gen, JS)
 accessor = mkPattern match
   where
-  -- WARN: if `prop` does not match the `IdentifierName` grammar, this will generate invalid code; see #2513
-  match (JSAccessor _ prop val) = Just (emit prop, val)
+  match (JSIndexer _ (JSStringLiteral _ prop) val) =
```

Comment:
I think it might be nice to provide a pattern synonym called `JSAccessor` which more or less does this - then I think that would be useful in e.g. the inliner.